src/github.com/stellar/gateway/compliance/handlers/request_handler_tx_status.go

```diff
@@ -0,0 +1,87 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/stellar/gateway/protocols"
+	"github.com/stellar/gateway/server"
+	"github.com/stellar/go/protocols/compliance"
+)
+
+// HandlerTxStatus implements /tx_status endpoint
+func (rh *RequestHandler) HandlerTxStatus(w http.ResponseWriter, r *http.Request) {
+
+	txid := r.URL.Query().Get("id")
+	if txid == "" {
+		log.Info("unable to get query parameter")
+		server.Write(w, protocols.NewMissingParameter("id"))
+		return
+	}
+	response := compliance.TransactionStatusResponse{}
+
+	if rh.Config.Callbacks.TxStatus == "" {
+		response.Status = compliance.TransactionStatusUnknown
+	} else {
+		endpoint := fmt.Sprintf(
+			"%s?id=%s",
+			rh.Config.Callbacks.TxStatus,
+			txid,
+		)
+
+		_, err := url.Parse(endpoint)
+		if err != nil {
+			log.Error(err, "failed to parse tx status endpoint")
+			server.Write(w, protocols.InternalServerError)
+			return
+		}
+		resp, err := rh.Client.Get(endpoint)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"tx_status": rh.Config.Callbacks.TxStatus,
+				"err":       err,
+			}).Error("Error sending request to tx_status server")
+			server.Write(w, protocols.InternalServerError)
+			return
+		}
+
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Error("Error reading tx_status server response")
+			server.Write(w, protocols.InternalServerError)
+			return
+		}
+
+		switch resp.StatusCode {
+		case http.StatusOK:
```

Comment:
Do we need `switch` here? Simple `if/else` would be simpler to read IMHO.

Reply:
Used switch here just in case we might want to evolve this is the future to check for different `http` responses.

Reply:
We can easily switch to `switch` later. You can leave it as it is now, it's not that important.